e2e/launchertester/basic_setup_test.go

```diff
@@ -80,9 +86,70 @@ func TestSetupWithCloudInit(t *testing.T) {
 			if tc.install_root {
 				args = append(args, "--root")
 			}
+
+			registrySet := make(chan error)
+			if !tc.withRegistry {
+				close(registrySet)
+			} else {
+				go func() {
+					defer close(registrySet)
+
+					d := gowsl.NewDistro(ctx, *distroName)
+					for {
+						select {
+						case <-ctx.Done():
+							registrySet <- ctx.Err()
+							return
+						default:
+						}
+
+						state, err := d.State()
+						if err != nil {
+							// WSL is not ready for concurrent access. Errors are very likely when registering or unregistering.
+							t.Logf("Couldn't to get distro state this time: %v", err)
+							time.Sleep(10 * time.Second)
+							continue
+						}
+
+						if state == gowsl.Uninstalling {
+							registrySet <- fmt.Errorf("too late: distro is uninstalling")
+							return
+						}
+
+						if state == gowsl.NonRegistered || state == gowsl.Installing {
+							t.Logf("Waiting for distro to be registered")
+							time.Sleep(10 * time.Second)
+							continue
+						}
+						// if running or stopped
+						id := wslCommand(ctx, "id", "-u", tc.wantUser)
+						out, err := id.CombinedOutput()
+						if err != nil {
+							t.Logf("Failed to get uid for %s: %v", tc.wantUser, err)
+							time.Sleep(300 * time.Millisecond)
+							continue
+						}
+						uid, err := strconv.Atoi(strings.TrimSpace(string(out)))
+						if err != nil {
+							t.Logf("Failed to parse %s: %v", out, err)
+							time.Sleep(300 * time.Millisecond)
+							continue
+						}
+						if uid < 1000 || uid > 60000 {
```

Comment:
hum, the condition seems to be linked to the previous case, no?

Reply:
Not exactly. For this test case we rely on cloud-init to create users and the cloud-config used is under control of the test case, so it seems reasonable to assume the UID we want to set from outside will be in that range.

Reply:
I would then set that this is a setup issue? To clarify it’s not something else that failed in the test itself.